### PR TITLE
created apiresult class to wrap all of the outputs from our api and upda...

### PR DESCRIPTION
**Pull request by severity**
...ted all of the public php files to use it, so far it seems to be working great, also added the ability to print out debug info when it is passed in to the class

### CODE REVIEW

File: migrations.sql [hunk 1]
```diff
@@ -77,6 +77,13 @@ CREATE OR REPLACE VIEW routelandia.highwaysHavingStations AS
   GROUP BY highways.highwayid
   HAVING count(distinct stations.stationid)>0;
 
+-------------------
```
Comment (joshproehl) [inline migrations.sql]: You did something here I'm pretty sure you didn't mean to do... You both moved and uncommented this function. :-)

File: entities/ApiResult.php [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace Routelandia\Entities;
+
+class ApiResult {
+	function __construct($input, $debug = null) {
+		is_array($input) ? $this->results = $input : $this->results = array($input);
+		is_null($debug) ? : $this->debug = $debug;
```
Comment (joshproehl) [inline entities/ApiResult.php]: This line was a little hard to read... 
I might have gone with something more like
```php
if(!is_null($debug) { $this->debug = $debug }
```

File: public/TrafficStats.php [hunk 3]
```diff
@@ -137,10 +138,11 @@ public function doPOST ($startpt, $endpt, $time,$request_data=null)
 
         $retVal = new stdClass;
         $retVal->query = $aboutQuery;
-        $retVal->info = $infoObj;
         $retVal->results = $qRes;
+        $retVal->info = $infoObj;
 
-        return $retVal;
+        return new ApiResult($retVal);
```
Comment (joshproehl) [inline public/TrafficStats.php]: Doesn't this create an object that looks like
```json
{
  "results": {
      "query": "q",
      "info": "i",
      "results": "r"
    }
}
```

We want to create the API result with the $qRes, and $aboutQuery in the debug spot, and then add the $infoObj on as a third value, THEN return the result.

**Comment by joshproehl** [comment]
@severity 3 comments to be addressed before we merge please. :-)

File: features/bootstrap/RestContext.php [hunk 4]
```diff
@@ -892,7 +1026,20 @@ public function theHighwayIsAHighway()
...
     {
-        return $this->detectorChecker($this->_data);
+    	$data = $this->_data;
+    	//var_dump(get_object_vars($data)); //for troubleshooting
+    	if (is_array($data->results)) {
+    		foreach ($data->results as $station)
+    		{
+    			if (!$this->detectorChecker($station))
+    			{
+    				throw new Exception("The station is not a station");
```
Comment (joshproehl) [inline features/bootstrap/RestContext.php]: :+1: great error message.

**Comment by severity** [comment]
@joshproehl: is there anything else I need to do to this branch or is it ready to be merged? :smiley:

File: features/bootstrap/RestContext.php (hunk 4 above)
Comment (severity) [inline features/bootstrap/RestContext.php]: lol, thanks :)

**Comment by joshproehl** [comment]
It looks pretty good! I'm looking at it locally to make sure, I'll merge it tonight or tomorrow morning. :-)

**Comment by severity** [comment]
Awesome, glad to hear it, sorry to bug you, ha ha!

**Comment by joshproehl** [comment]
Merged at 1ea021fbb164ff5914ef9c1d6eeac14a622bb8f6

**Pull request closed without merging**